### PR TITLE
ci: use go1.23

### PR DESCRIPTION
**Pull request by rickystewart**
Part of: CRDB-41921

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -132,7 +132,7 @@ endif
 
 .PHONY: format
 format:
-	go install -C internal/devtools github.com/cockroachdb/crlfmt && crlfmt -w -tab 2 ../..
```
Comment (rickystewart) [inline Makefile:135]: I was pretty perplexed by this. To me this `../..` seems transparently obviously wrong and it should be `.`. This change fixes CI ([link](https://github.com/cockroachdb/pebble/actions/runs/11222624774/job/31195468780?pr=4012)).

Comment (RaduBerinde) [reply Makefile:135]: Indeed, thanks for the fix. I think there was an intermediate version that used `go run` and the working dir would be `./internal/devtools`.

**Comment by RaduBerinde** [comment]
Do we plan to switch to go 1.23 for 24.2.3? If not, we should either hold off on this, or test both 1.22 and 1.23 in CI

**Comment by rickystewart** [comment]
Yes, we plan to switch to Go 1.23 for 24.3.

**Review by RaduBerinde — APPROVED** [review]
Thanks!

**Pull request merged**